nilearn/decoding/decoder.py

```diff
@@ -476,11 +474,16 @@ def fit(self, X, y, groups=None):
         else:
             n_problems = 1
 
+        # Check if the size of the mask image and the number of features allow
+        # to perform feature screening.
+        selector = check_feature_screening(self.screening_percentile,
+                                           self.mask_img_,
+                                           self.is_classification)
         # Return a suitable screening percentile according to the mask image
-        self.screening_percentile_ = _adjust_screening_percentile(
-            self.screening_percentile, self.mask_img_, verbose=self.verbose)
+        if hasattr(selector, 'percentile'):
+            self.screening_percentile = selector.percentile
```

Comment:
should this be `self.screening_percentile_` ?

Reply:
maybe give it a value in any case, eg 100 or None by default?

Reply:
> maybe give it a value in any case, eg 100 or None by default?

But I thought the default value is already 20?
 
> should this be `self.screening_percentile_` ?

Indeed, I will make a quick fix

Reply:
I mean in the case that `selector` doesn't have a `percentile` attribute, maybe the decoder should still have one right? so that its interface is always the same

Reply:
```diff
modified   nilearn/decoding/decoder.py
@@ -481,9 +481,11 @@ class _BaseDecoder(LinearModel, RegressorMixin, CacheMixin):
                                            self.is_classification)
         # Return a suitable screening percentile according to the mask image
         if hasattr(selector, 'percentile'):
-            self.screening_percentile = selector.percentile
+            self.screening_percentile_ = selector.percentile
+        else:
+            self.screening_percentile_ = self.screening_percentile
 
-        n_final_features = int(X.shape[1] * self.screening_percentile
+        n_final_features = int(X.shape[1] * self.screening_percentile_
                                * self.clustering_percentile / 10000)
         if n_final_features < 50:
             warnings.warn(
```
Agree, something like this?

Reply:
yes, just make sure `self.screening_percentile` is never None or the `n_final_features` computation will fail. actuall chec_screening_percentile will only return None if screening_percentile is 100 or None, so you could just write `self.screening_percentile_ = 100` -- or handle the `None` case in any other way you see fit